src/CoreWCF.Kafka.Client/src/CoreWCF/ServiceModel/Channels/Helpers/KafkaChannelHelpers.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Globalization;
+using System.ServiceModel;
+using Confluent.Kafka;
+
+namespace CoreWCF.ServiceModel.Channels;
+
+internal static class KafkaChannelHelpers
+{
+    public static Exception ConvertProduceException(ProduceException<Null, byte[]> produceException)
+    {
+        return new CommunicationException(
+            string.Format(CultureInfo.CurrentCulture,
```

Comment:
Is there any way to detect a timeout scenario. This would happen if the CancellationToken is cancelled. In that scenario, we would normally throw TimeoutException. It's the only exception we throw regularly which isn't derived from CommunicationException